src/unistd.rs

```diff
@@ -578,6 +562,55 @@ pub fn chown<P: ?Sized + NixPath>(path: &P, owner: Option<Uid>, group: Option<Gi
     Errno::result(res).map(drop)
 }
 
+/// Change the ownership of the file at `path` to be owned by the specified
+/// `owner` (user) and `group` (see
+/// [lchown(2)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/lchown.html)).
+/// If `path` names a symbolic link, then ownership of the symbolic link file itself is changed.
+///
+/// The owner/group for the provided path name will not be modified if `None` is
+/// provided for that argument.  Ownership change will be attempted for the path
+/// only if `Some` owner/group is provided.
+#[inline]
+pub fn lchown<P: ?Sized + NixPath>(path: &P, owner: Option<Uid>, group: Option<Gid>) -> Result<()> {
+    let res = try!(path.with_nix_path(|cstr| {
+        // According to the POSIX specification, -1 is used to indicate that
+        // owner and group, respectively, are not to be changed. Since uid_t and
+        // gid_t are unsigned types, we use wrapping_sub to get '-1'.
+        unsafe { libc::lchown(cstr.as_ptr(),
+                              owner.map(Into::into).unwrap_or((0 as uid_t).wrapping_sub(1)),
+                              group.map(Into::into).unwrap_or((0 as gid_t).wrapping_sub(1))) }
+    }));
+
+    Errno::result(res).map(drop)
+}
+
+/// Change the ownership of the file specified by the file descriptor `fd` to be owned by the
+/// specified `owner` (user) and `group` (see
+/// [fchown(2)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fchown.html)).
+///
+/// The owner/group for the provided path name will not be modified if `None` is
+/// provided for that argument.  Ownership change will be attempted for the path
+/// only if `Some` owner/group is provided.
+#[inline]
+pub fn fchown(fd: RawFd, owner: Option<Uid>, group: Option<Gid>) -> Result<()> {
+    let res = unsafe { libc::fchown(fd,
+                             owner.map(Into::into).unwrap_or((0 as uid_t).wrapping_sub(1)),
```

Comment:
You should probably wait until #955 lands, then rebase this PR and use the private `chown_raw_ids` method here.

Reply:
I would do so if I still have spare time by the time #955 lands.

Reply:
Done.